packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -926,7 +929,27 @@ export function getValidatorApi(
       }
 
       if (!state) {
-        state = await chain.getHeadStateAtCurrentEpoch(RegenCaller.getDuties);
+        if (epoch >= currentEpoch - 1) {
```

Comment:
should be this?
```suggestion
        if (epoch >= currentEpoch) {
```

Reply:
yes, the cached beacon state stores proposers for prev, current and next epoch, which means if request epoch is within that range, we can just use the head state at current epoch

Reply:
added a comment in https://github.com/ChainSafe/lodestar/pull/7130/commits/f85b86e91dcfef3a2e8c5683474c0f3e1e84fc66 to clarify this